### PR TITLE
Doc(eos_designs): Update custom_structured_config doc following merge changes in 4.0.0

### PR DESCRIPTION
**Pull request by alexeygorbunov**
## Change Summary

Update documentation related to custom_structured_config feature following the change of the default merge strategy from `replace` to `append_rp` in v4.0.0

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes
- Update examples and related notes

## How to test
Use default `append_rp` merge strategy to confirm interaction between:
- `eos_designs`'s `name_servers` vs `eos_cli_config_gen`'s `ip_name_servers`
- `eos_designs`'s `ipv4_acls` vs `eos_cli_config_gen`'s `ip_access_lists`

### User Checklist
- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/custom-structured-configuration.md [hunk 1]
```diff
@@ -68,20 +68,20 @@ ip name-server vrf MGMT 192.168.42.40
...
 
 ```eos
-ip name-server vrf EOS_CLI 4.4.4.4
-ip name-server vrf EOS_CLI 8.8.8.8
+ip name-server vrf EOS_CLI 1.1.1.1
+ip name-server vrf EOS_CLI 192.168.42.10
 ```
 
-#### eos_cli_config_gen variables overwritten by eos_designs variables
+#### eos_cli_config_gen variables and eos_designs variables
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/roles/eos_designs/docs/how-to/custom-structured-configuration.md]: The body still says that eos_cli_config_gen variables overwritten by eos_designs variables so can we keep the heading same.

Comment (alexeygorbunov) [reply ansible_collections/arista/avd/roles/eos_designs/docs/how-to/custom-structured-configuration.md]: Fixed

**Comment by ClausHolbechArista** [comment]
Please prioritize to get this PR ready if we need it in 5.0.

**Review by ClausHolbechArista — APPROVED** [review]

File: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/custom-structured-configuration.md [hunk 2]
```diff
@@ -105,59 +105,119 @@ ip name-server vrf MGMT 192.168.42.10
...
+  - ip_address: 192.168.42.10
+    vrf: EOS_CLI
 ```
 
 will generate as intended config:
 
 ```eos
-ip name-server vrf CUSTOM_STRUCT 1.1.1.1
-ip name-server vrf CUSTOM_STRUCT 2.2.2.2
+ip name-server vrf EOS_CLI 1.1.1.1
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_designs/docs/how-to/custom-structured-configuration.md]: Let's update this IP address to a private IP.

Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_designs/docs/how-to/custom-structured-configuration.md]: Disregard 1.1.1.1 is a public DNS resolver operated by Cloudflare so we can keep this!

Comment (alexeygorbunov) [reply ansible_collections/arista/avd/roles/eos_designs/docs/how-to/custom-structured-configuration.md]: Just to be on a safe side - converted all Public IPs not owned by Arista to private ones

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Pull request merged**